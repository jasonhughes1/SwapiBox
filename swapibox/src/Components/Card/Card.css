.card-holder {
  background-color: #A0A469;
  width: 500px;
  margin: 10px;
  border-radius: 25px;
  justify-content: center;
  flex-direction: column;
  box-shadow: 10px 5px 5px grey; }

.card-holder:hover {
  background-color: grey;
  cursor: pointer; }
