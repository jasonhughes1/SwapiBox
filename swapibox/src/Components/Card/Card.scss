.card-holder {
  border: 2px solid yellow;
  width: 400px;
  height: 500px;
  margin: 10px;
  border-radius: 25px;
  display: flex;
  justify-content: center;
  flex-direction: column;
}
