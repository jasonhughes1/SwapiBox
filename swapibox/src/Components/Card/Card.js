import React from 'react';
import './Card.css';

const Card = ({ cardData, setFavorite }) => {
  const cardKeys = Object.keys(cardData);
  const cards = cardKeys.map((key, index) => {
  const populateCard = cardData[dataReturn(key)];

    function dataReturn(key) {
      if (key === 'Residents') {
        cardData.Residents =
        cardData.Residents.map(resident => resident.name + ' ');
        return key;
      } else {
        return key;
      }
    }
    return (
      <div
        key={index}>
        <h3>{key}</h3>
         <p>{populateCard}</p>
      </div>
    );
  });

  return (
    <article className='card-holder'>
      <button className='favorite-btn'
        onClick={() => setFavorite(cardData)}>Favorite</button>
    { cards }
    </article>
  );
};


export default Card;
