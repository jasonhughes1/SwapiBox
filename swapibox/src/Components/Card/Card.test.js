import React from 'react';
import Card from './Card.js';
import { shallow } from 'enzyme';


const mockSetFavorite = jest.fn();
let renderedCard;

describe('Card', () => {
  beforeEach(() => {
    renderedCard = shallow(
      <Card
        cardData={{card: 1}}
        setFavorite={mockSetFavorite}
      />);
  });

  it('should render corrrectly', () => {
    expect(renderedCard).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(renderedCard).toMatchSnapshot();
  });
});
