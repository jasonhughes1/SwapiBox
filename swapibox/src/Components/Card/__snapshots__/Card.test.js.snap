// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardData={
        Object {
            "card": 1,
          }
    }
    setFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="favorite-btn highlight"
          onClick={[Function]}
>
          Favorite
</button>,
        Array [
          <div>
            <h3>
                        card
            </h3>
            <p>
                        1
            </p>
</div>,
        ],
      ],
      "className": "card-holder",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorite",
          "className": "favorite-btn highlight",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Favorite",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              card
</h3>,
            <p>
              1
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "card",
            },
            "ref": null,
            "rendered": "card",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 1,
            },
            "ref": null,
            "rendered": 1,
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="favorite-btn highlight"
            onClick={[Function]}
>
            Favorite
</button>,
          Array [
            <div>
              <h3>
                            card
              </h3>
              <p>
                            1
              </p>
</div>,
          ],
        ],
        "className": "card-holder",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite",
            "className": "favorite-btn highlight",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Favorite",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                card
</h3>,
              <p>
                1
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "card",
              },
              "ref": null,
              "rendered": "card",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
